src/main/java/io/github/eocqrs/cmig/sha/Sha.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright (c) 2023 Aliaksei Bialiauski, EO-CQRS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.github.eocqrs.cmig.sha;
+
+import io.github.eocqrs.cmig.meta.Names;
+import org.cactoos.Scalar;
+import org.cactoos.bytes.Sha256DigestOf;
+import org.cactoos.io.InputOf;
+import org.cactoos.text.HexOf;
+
+import java.util.List;
+
+/**
+ * SHA256 for State.
+ *
+ * @author Aliaksei Bialiauski (0.0.0)
+ * @since 0.0.0
+ */
+public final class Sha implements Scalar<String> {
```

Comment:
I think this class is better implemented as `Text`, wdyt?

Reply:
more flexible `Sha` implementation, can encapsulate, for example, `CmigState` and turn 1 state into 1 sha, wdyt?

Reply:
@l3r8yJ great idea, but now I'm not sure how to design `State` interface, that's why I created puzzle for it